pkg/tsdb/mssql/mssql.go

```diff
@@ -65,11 +65,17 @@ func newInstanceSettings(cfg *setting.Cfg) datasource.InstanceFactoryFunc {
 		if err != nil {
 			return nil, fmt.Errorf("error reading settings: %w", err)
 		}
+
+		database := settings.Database
```

Comment:
Maybe the opposite, check the jsondata first and then fallback to legacy field?

Reply:
Changed it.